dask_expr/_collection.py

```diff
@@ -403,6 +403,25 @@ def __getitem__(self, other):
     def __dask_tokenize__(self):
         return type(self).__name__, self._expr._name
 
+    def __repr__(self):
+        data = self._repr_data().to_string(max_rows=5)
+        _str_fmt = """Dask {klass} Structure:
+{data}
+Dask Name: {name}, {layers}
+Expr={expr}"""
+        if len(self.columns) == 0:
+            data = data.partition("\n")[-1].replace("Index", "Divisions")
+            _str_fmt = f"Empty {_str_fmt}"
+        layers = len({k for (k, *_) in self.dask.keys()})
```

Comment:
This  still has the old version that materialises the graph?

Reply:
ya, sorry, requested a review too so. My apologies.